File: 2099-number-of-strings-that-appear-as-substrings-in-word/number-of-strings-that-appear-as-substrings-in-word.py
#Hello Ji Hello
class Solution:
    def numOfStrings(self, patterns: list[str], word: str) -> int:
        count = 0
        
        # Loop through each individual pattern string in the list
        for pattern in patterns:
            # Check if the pattern exists as a substring within 'word'
            if pattern in word:
                count += 1 # Increment the counter if a match is found
                
        # Return the total number of matching patterns
        return count

File: 2099-number-of-strings-that-appear-as-substrings-in-word/number-of-strings-that-appear-as-substrings-in-word.java
class Solution {
    public int numOfStrings(String[] patterns, String word) {
        int count = 0;
        
        // Loop through each individual pattern string in the array
        for (String pattern : patterns) {
            // Check if the 'word' contains the current pattern as a substring
            if (word.contains(pattern)) {
                count++; // Increment the counter if a match is found
            }
        }
        
        // Return the total number of matching patterns
        return count;
    }
}